kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -0,0 +1,165 @@
+const CoreApp = require('kolibri');
+const ConditionalPromise = require('kolibri.lib.conditionalPromise');
+const CoreActions = require('kolibri.coreVue.vuex.actions');
+const GetDefaultChannelId = require('kolibri.coreVue.vuex.getters').getDefaultChannelId;
+const Constants = require('../../constants');
+
+const ClassroomResource = CoreApp.resources.ClassroomResource;
+const ChannelResource = CoreApp.resources.ChannelResource;
+const LearnerGroupResource = CoreApp.resources.LearnerGroupResource;
+
+
+function _classState(classroom) {
+  return {
+    id: classroom.id,
+    name: classroom.name,
+  };
+}
+
+function _classesState(classes) {
+  return classes.map(classroom => _classState(classroom));
+}
+
+function _channelState(channel) {
+  return {
+    id: channel.id,
+    name: channel.name,
+    root_pk: channel.root_pk,
+  };
+}
+
+function _channelsState(channels) {
+  return channels.map(channel => _channelState(channel));
+}
+
+function _groupState(group) {
+  return {
+    id: group.id,
+    name: group.name,
+  };
+}
+
+function _groupsState(groups) {
+  return groups.map(group => _groupState(group));
+}
+
+function _getCurrentChannelObject(currentChannelId, channels) {
+  return channels.find(channel => channel.id === currentChannelId);
+}
+
+function displayModal(store, modalName) {
+  store.dispatch('SET_MODAL', modalName);
+}
+
+
+function showExamsPage(store, classId) {
```

Comment:
might be best to restructure this using sub page model https://github.com/learningequality/kolibri/pull/1117/files#r108287946